addons/im_livechat/models/mail_channel.py

```diff
@@ -89,6 +89,14 @@ def _notify_typing_partner_data(self):
             data['livechat_username'] = self.env.user.partner_id.user_livechat_username
         return data
 
+    def _channel_seen(self, last_message_id=None):
+        res = super(MailChannel, self)._channel_seen(last_message_id)
+        if self.channel_type == 'livechat':
+            self.channel_last_seen_partner_ids.filtered(
+                lambda cp: cp.is_pinned and cp.partner_id == self.env.user.partner_id
+            ).write({'last_seen_dt': fields.Datetime.now()})
+        return res
```

Comment:
This could be moved into the base code directly, no need to limit it to livechat channels.

Reply:
I think we should exclude normal channels, because if we unpin them JS will automatically unsubscribe `user` from the channel.

Reply:
The update of the `last_seen_dt` should be in the base code, but indeed the garbage collect should be specific to livechat channels.

Reply:
Should we do this for 1-1 DM Chats also?

Reply:
I think we can just do it all the time, there is already a query to update the `fetched_message_id`/`seen_message_id` we might as well always add the date into the same update in the base code.